src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,85 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int INITIAL_CAPACITY = 16;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
-
+        if (size >= table.length * LOAD_FACTOR) {
+            resize();
+        }
+        Node<K, V> currentNode = table[hash(key)];
+        if (currentNode == null) {
+            table[hash(key)] = new Node<>(key, value, null);
+        }
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+                return;
+            }
+            if (currentNode.next == null) {
+                currentNode.next = new Node<>(key, value, null);
+                break;
+            }
```

Comment:
useless block. just create new node on the line 34

Reply:
If the block is useless, why would the tests fail without it?
I don’t understand how to create a new node without this block to make the tests successful.

Reply:
The possible solution

```
Node<K, V> currentNode = table[hash(key)];
        if (currentNode == null) {
            table[hash(key)] = new Node<>(key, value, null);
            size++;
            return;
        }
        if (Objects.equals(currentNode.key, key)) {
            currentNode.value = value;
            return;
        }
        while (currentNode.next != null) {
            currentNode = currentNode.next;
            if (Objects.equals(currentNode.key, key)) {
                currentNode.value = value;
                return;
            }
        }
        currentNode.next = new Node<>(key, value, null);
        size++;
```

But I am not sure which variant is better. I think your variant also quite good.